pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -250,18 +266,19 @@ func (r *Reconciler) fetchAllResourcesInOneNamespace(_ context.Context, namespac
 			continue
 		}
 		if !r.InformerManager.IsInformerSynced(gvr) {
-			return nil, fmt.Errorf("informer cache for %+v is not synced yet", gvr)
+			return nil, controller.NewInformerError(fmt.Errorf("informer cache for %+v is not synced yet", gvr))
 		}
 		lister := r.InformerManager.Lister(gvr)
 		objs, err := lister.ByNamespace(namespaceName).List(labels.Everything())
 		if err != nil {
-			return nil, fmt.Errorf("cannot list all the objects of type %+v in namespace %s: %w", gvr, namespaceName, err)
+			return nil, controller.NewInformerError(fmt.Errorf("cannot list all the objects of type %+v in namespace %s: %w", gvr, namespaceName, err))
```

Comment:
This is where it gets puzzling. This error is a typically read error, while the one above (269) is an initialization error which indicates that the handler hasn't received all pre-sync related updates. Should they be considered of the same category?